pages/chaoss/visualizations/contrib_activity_cycle.py

```diff
@@ -0,0 +1,178 @@
+                                                {
+                                                    "label": "Weekday",
+                                                    "value": "D",
+                                                },
+                                                {"label": "Hourly", "value": "H"},
+                                            ],
+                                            value="D",
+                                            inline=True,
+                                        ),
+                                    ]
+                                ),
+                                dbc.Col(
+                                    dbc.Button(
+                                        "About Graph",
+                                        id=f"{PAGE}-popover-target-{VIZ_ID}",
+                                        color="secondary",
+                                        size="sm",
+                                    ),
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ],
+                            align="center",
+                        ),
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+# callback for graph info popover
+@callback(
+    Output(f"{PAGE}-popover-{VIZ_ID}", "is_open"),
+    [Input(f"{PAGE}-popover-target-{VIZ_ID}", "n_clicks")],
+    [State(f"{PAGE}-popover-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+# callback for VIZ TITLE graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(VIZ_ID + "-interval", "value"),
+    ],
+    background=True,
+)
+def contrib_activity_cycle_graph(repolist, interval):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cmq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cmq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, interval)
+
+    fig = create_figure(df, interval)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, interval):
+
+    # for this usecase we want the datetimes to be in their local values
+    # tricking pandas to keep local values when UTC conversion is required for to_datetime
+    df["author_timestamp"] = df["author_timestamp"].astype("str").str[:-6]
+    df["committer_timestamp"] = df["author_timestamp"].astype("str").str[:-6]
```

Comment:
I think that these are both going to be 'author_timestamp' rather than one being 'committer_timestamp'

Reply:
since committer and author timestamps could be the same (and often are) you will double count if you concatenate them together. if they're not equal you should include both, if they are you should include only one. that means they're an author-committer.